openff/toolkit/topology/topology.py

```diff
@@ -1810,10 +1810,10 @@ def from_pdb(
         )
 
         for off_atom, atom in zip([*topology.atoms], pdb.topology.atoms()):
-            off_atom.metadata["residue_name"] = atom.residue.name
-            off_atom.metadata["residue_number"] = atom.residue.id
-            off_atom.metadata["insertion_code"] = atom.residue.insertionCode
-            off_atom.metadata["chain_id"] = atom.residue.chain.id
+            off_atom.metadata["residue_name"] = atom.residue.name # type:ignore[attr-defined]
+            off_atom.metadata["residue_number"] = atom.residue.id # type:ignore[attr-defined]
+            off_atom.metadata["insertion_code"] = atom.residue.insertionCode # type:ignore[attr-defined]
+            off_atom.metadata["chain_id"] = atom.residue.chain.id # type:ignore[attr-defined]
```

Comment:
Thought a little bit about how to make this cleaner, could only come up with using `getattr` which isn't actually any cleaner. Remind me that all of these attributes being optional is (intentionally) part of the design?